monomer.go

```diff
@@ -38,6 +38,11 @@ type Application interface {
 	RollbackToHeight(uint64) error
 }
 
+type DuplexAdapter struct {
+	CosmosToEth CosmosTxAdapter
+	EthToCosmos PayloadTxAdapter
+}
+
```

Comment:
Is this a convenience type? I'm not understanding its purpose.

Reply:
Just convenience - grouping similar functions.

Reply:
Personally, I'd prefer either putting them as receivers on the same type or leaving them entirely separate. Having functions as fields on a struct is generally confusing imho (the `SelectiveListener`s being the only useful case I can think of).

Reply:
Removed the grouping struct here: be386bb